### PR TITLE
editor: set the focus in the forms

### LINKED ISSUE #542: Set the focus in the forms.
**Describe the issue**
As a user opens a form, ie the editor, the focus is not set.

**To Reproduce**
1. Log in as a librarian.
1. Try to create a new resource, for example a new circulation policy.
1. Assert that you need multiple tab key hits, or to use the mouse, to type something into the first input of the form.

**Expected behavior**
Focus should be set where it's the most probable that a user wants to start to type.

### PR DESCRIPTION
**Pull request by lauren-d**
* Adds angular directive.
* Updates remote-input component to add autofocus binding.
* Sets the focus in the forms.
* Updates form options to set default focus.
* Closes #542.

Co-Authored-by: Benoit Erken <erken.benoit@gmail.com>
Co-Authored-by: Lauren-D <laurent.dubois@itld-solutions.be>

## Why are you opening this PR?

- Implements task https://tree.taiga.io/project/rero21-reroils/task/1117.
- Fix issues #542.

## How to test?

- Log in as a librarian.
- Try to create a new resource, for example a new circulation policy.

**note:** It may be necessary to clear the browser cache

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?

### CODE REVIEW

**Comment by jma** [comment]
Please use `rero-ils-ui` instead of modifying legacy ui dir of the `rero-ils`.

File: rero_ils/modules/locations/jsonschemas/form_locations/location-v0.0.1.json [hunk 1]
```diff
@@ -14,7 +14,8 @@
     "type": "fieldset",
     "items": [
       {
-        "key": "name"
+        "key": "name",
+        "autofocus": "true"
```
Comment (jma) [inline rero_ils/modules/locations/jsonschemas/form_locations/location-v0.0.1.json]: As JSON support boolean, I would replace the string `"true"` by a boolean `true`. You can fix it to the other files.

File: ui/src/app/records/custom-editor/circulation-settings/circulation-policy/circulation-policy.component.html [hunk 2]
```diff
@@ -26,8 +26,14 @@ <h1 translate>Circulation Policy</h1>
     <div class="form-group row">
       <label for="name" class="col-sm-2 col-form-label required" translate>Name</label>
       <div class="col-sm-10">
-        <input class="form-control" id="name" formControlName="name"
-          placeholder="{{ 'Please insert a name' | translate }}" required>
+        <input
```
Comment (jma) [inline ui/src/app/records/custom-editor/circulation-settings/circulation-policy/circulation-policy.component.html]: I will do the review for all ui file in an other PR to the `rero-ils-ui` project.

File: rero_ils/modules/locations/jsonschemas/form_locations/location-v0.0.1.json (hunk 1 above)
Comment (lauren-d) [reply rero_ils/modules/locations/jsonschemas/form_locations/location-v0.0.1.json]: done

**Pull request closed without merging**